qadence/serialization.py

```diff
@@ -43,27 +47,127 @@ def file_extension(file: Path | str) -> str:
     AbstractBlock,
     QuantumCircuit,
     QuantumModel,
-    QNN,
-    TransformedModule,
     Register,
-    Basic,
+    core.Basic,
     torch.nn.Module,
 ]
 SUPPORTED_TYPES = TypingUnion[
     AbstractBlock,
     QuantumCircuit,
     QuantumModel,
-    QNN,
-    TransformedModule,
     Register,
-    Basic,
+    core.Basic,
     torch.nn.Module,
 ]
 
+EXPR_TYPES = TypingUnion[core.Expr, float, int, str, tuple, dict]
```

Comment:
Use pipe to type unionise ?

Reply:
I will get rid of this one since it's not used anymore. The `SUPPORTED_TYPES` uses `TypingUnion`, should we replace for something else?